Automattic/simplenote-ios
# Dropping VSTheme: Mark II

### Fix
This PR is part of a multistep approach at removing VSTheme:

- **EditorViewController:** Dropped VSTheme usage
- **EditorViewController:** Swifted `refreshStyle` + split into smaller methods
- **TextStorage:** Explicit properties for Headline / Body attributes
- **New Extensions**: UIDevice / UIKeyboardAppearance / NSMutableParagraph
- Dropped several unneeded imports

Since I was there, I've noticed that the Editor's Typing Attributes were off: when there is no text, we should use the Header Typing Attributes. This causes the cursor to show up with the right size (in empty notes).

@eshurakov Sorry to bug you yet again!!!
(And thanks in advance!!)

Ref. #481

---

### Test: Typing Attributes / Storage
1. Add a new note

- [x] Verify while the note is Empty, the Cursor's height matches the Header font
- [x] Verify that while you type a Note Title, the cursor's height is preserved
- [x] Verify that the second you press `<return>` the cursor assumes the Body Font Height

### Test: Editor Style
1. Launch the app
2. Press over any note to open the Editor

- [x] Verify the background color matches the selected theme
- [x] Verify the Keybaord matches the selected theme
- [x] Verify that newly added Checklists look great
- [x] Verify the Header / Body text looks great
- [x] Please repeat for Dark / Light / System themes!

### Test: Previewing
1. Launch the app
2. Toggle Dark Mode
3. Long press over any note in your list

- [x] Verify that the Previewing Window looks slightly gray
- [x] Verify that pressing over the preview causes the Editor to be pushed
- [x] Verify that the Editor is pitch black (rather than gray!)

### Test: Line Spacing
1. Launch the app
2. Press over any note to open the Editor

- [x] Verify Interline Spacing looks great on iPhone devices
- [x] Verify Interline Spacing looks great on iPad devices

### Release
These changes do not require release notes.
